src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/ProfileResolver.java

```diff
@@ -134,8 +134,14 @@ public void resolve(@NotNull ResolutionData data) throws IOException, IllegalSta
 
     Metadata metadata = new Metadata();
     metadata.setTitle(profile.getMetadata().getTitle());
-    metadata.setVersion(profile.getMetadata().getVersion());
-    metadata.setOscalVersion(OscalUtils.OSCAL_VERSION);
+
+    if (profile.getMetadata().getVersion() != null) {
+      metadata.setVersion(profile.getMetadata().getVersion());
+    }
+
+//    metadata.setOscalVersion(OscalUtils.OSCAL_VERSION);
+    metadata.setOscalVersion(data.getProfile().getMetadata().getOscalVersion());
```

Comment:
OK, you will love this feedback in the PR. I decided to read the [profile resolution spec](https://github.com/usnistgov/OSCAL/blob/adc51808988404448fddc9b85da7c5084e6a03af/src/specifications/profile-resolution/profile-resolution-specml.xml#L1391-L1393) to make sure I understood this bit a few times.

> The value of metadata:oscal-version in the target MUST be set with a string that identifies the version of OSCAL used by this tool to resolve the profile (ex. 1.0.0). This value MUST be determined by compiling the oscal-versions from each imported document and the source profile, and taking the most recent minor version. If this version is more recent than what the resolution tool is using, then the value of oscal-version MUST be the version that the tool used internally. If any of the above OSCAL versions (imported document versions, source profile version, tool version) are of a different major version (the first digit differs), the tool SHOULD provide an error and cease processing.

What you have done is definitely good, but we specify _minor_ version, not _patch_ version `currently in this PR the constant is set to `1.0.4.`.

Is this a specification issue or a code issue?

Reply:
I believe this is a spec issue. Would you open an issue on the spec?

Reply:
https://github.com/usnistgov/OSCAL/issues/1272